theano/gof/destroyhandler.py

```diff
@@ -793,14 +794,14 @@ def fast_destroy(self, app):
         dm = getattr(app.op, 'destroy_map', None)
         if not dm:
             return
-        # inputs = sum(dm.values())  # list of app's destroyed inputs
-        inputs = dm.values()[0]
+        inputs = list(set(itertools.
+                          chain.from_iterable(dm.values())))   # list of app's destroyed inputs
```

Comment:
Is the problem that in python 3 dm.values() is an iterable? If so, this would be a simpler fix:

    inputs = sum(list(dm.values))